### ISSUE TITLE
GoMetaLinter can't find directory of a failing package from sub-directory tree

### ISSUE THREAD

**Issue by mirza-s** [fatih/vim-go#1417]
### Behavior
GoMetaLinter can't find directory of a failing package (in sub-directory tree)

The following error is shown:
```
"test2/test2.go" [New DIRECTORY]
Error detected while processing function <SNR>13_template_autocreate[3]..go#template#create:
line    6:
E344: Can't find directory "test2" in cdpath
E472: Command failed
Press ENTER or type command to continue
```

GoMetaLinter should locate the failing file with no error.

### Steps to reproduce:

1. Checkout `https://github.com/mirza-s/repro` into `$GOPATH/src/repro`
2. From `$GOPATH/src/repro` execute `vi test/test.go`
3. Run `:GoMetaLinter`

### Configuration

* `vimrc` you used to reproduce
```
  execute pathogen#infect()
  syntax on
  filetype plugin indent on
  set number
  set ignorecase
  set smartcase
```
* vim version: 
```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Aug 22 2017 06:09:48)
MacOS X (unix) version
Included patches: 1-983
Compiled by Homebrew
```
* vim-go version: `commit 25d00acf3958cc3f0e1402ab1cef8afe817cb1ae (HEAD, tag: v1.14)`
* go version: `go version go1.9 darwin/amd64`

**Comment by mirza-s** [comment]
I have bisected this to commit 23b7c22686a501629146d39e3c1ac02298f8d5c4
Before this commit the error is:

```
[Quickfix List] :    cgetexpr a:items                                                                                                                    1,1             All
"test2/test2.go" [New DIRECTORY]
```

**Comment by Carpetsmoker** [comment]
Fixed via #1414.
